src/main.rs

```diff
@@ -1366,13 +1247,15 @@ fn parse_manpage_at_path(
     let mut parsers = parsers.peekable();
 
     if parsers.peek().is_none() {
-        // add_diagnostic(format!("{}: Not supported", manpage_path));
+        tracing::info!("{}: Not supported", manpage_path.display());
     }
 
-    if let Some(mut completions) = parsers
-        // .inspect(|parser| add_diagnostic(format!("Trying {}", parser)))
-        .find_map(|parser| parser.parse_man_page(&manpage, &cmdname))
-    {
+    if let Some(mut completions) = parsers.find_map(|parser| {
+        // Second (last) level span
+        let span = tracing::info_span!("Trying", "{}", parser.to_string().as_str());
```

Comment:
```suggestion
        let span = tracing::info_span!("Trying", "{}", parser);
```

Reply:
Nice catch, I didn't check the blocks a lot when I was doing modification.